src/templates/Salons/index.tsx

```diff
@@ -0,0 +1,57 @@
+import { Button, Stack, Typography } from '@mui/material';
+import { ReactNode } from 'react';
+import { useTranslation } from 'react-i18next';
+import { Box } from '@mui/system';
+import { useLocation, Redirect } from 'react-router-dom';
+import Spinner from 'components/Spinner';
+import { useFetchAllSalonsQuery } from 'store/api/admin';
+import { CustomEditHeader, CustomSalonsNavigation } from './salons.styled';
+import NavTabs from './salons.navtabs';
+import SalonsNavigation from 'components/AdminPanelNavigation/adminPanelNavigation.list';
+
+const SalonsTemplate = ({ children }: { children: ReactNode }) => {
+  const [translation] = useTranslation();
+  const { data = [], isFetching, isError } = useFetchAllSalonsQuery();
+  const { pathname } = useLocation();
+
+  if (isError)
+    return (
+      <Typography variant="h5" textAlign="center" marginTop={5}>
+        {translation('admin:salonTemplate.error')}
+      </Typography>
+    );
+
+  if (isFetching) return <Spinner />;
+
+  if (pathname === '/salons') return <Redirect to={`/salons/${data[0]._id || 1}/edit`} />;
+
+  return (
+    <Stack direction="row">
+      <CustomSalonsNavigation p={2}>
+        <SalonsNavigation data={data} />
+      </CustomSalonsNavigation>
+      <Box width="100%" display="flex" flexDirection="column" minHeight="400px">
```

Comment:
It is not required to use width or height everywhere. In that case, `minHeight` is needless and `width` can be replaced by `flexGrow`. Thanks to that, the application will be more responsive in the default way.